pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -142,15 +149,17 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			// If the SubnetPort CR's status changed, let's clean the conditions, to ensure the r.Client.Status().Update in the following updateSuccess will be invoked at any time.
 			subnetPort.Status.Conditions = nil
 		}
-		r.StatusUpdater.UpdateSuccess(ctx, subnetPort, setReadyStatusTrue, subnetPort, r.SubnetPortService)
+		r.StatusUpdater.UpdateSuccess(ctx, subnetPort, setReadyStatusTrue, r.SubnetPortService)
 	} else {
 		r.StatusUpdater.IncreaseDeleteTotal()
 		subnetPortID := r.SubnetPortService.BuildSubnetPortId(&subnetPort.ObjectMeta)
 		if err := r.SubnetPortService.DeleteSubnetPortById(subnetPortID); err != nil {
 			r.StatusUpdater.DeleteFail(req.NamespacedName, nil, err)
+			setAddressBindingStatusBySubnetPort(r.Client, ctx, subnetPort, r.SubnetPortService, metav1.Now(), subnetPortRealizationError)
 			return common.ResultRequeue, err
 		}
 		r.StatusUpdater.DeleteSuccess(req.NamespacedName, nil)
+		setAddressBindingStatusBySubnetPort(r.Client, ctx, subnetPort, r.SubnetPortService, metav1.Now(), vmOrInterfaceNotFoundError)
```

Comment:
Do we need consider the case kube-apiserver received VM delete event but no SubnetPort delete event（negative case）？

Reply:
No, actually AddressBinding CR is a sub-resource of SubnetPort CR. If we want to support this, we should process SubnetPort CR first.